### PR TITLE
Add new querystring package to escape special characters

### PR DESCRIPTION
**Pull request by joaofranciscosantos**
This is related to issue #1040. I noticed there is also a PR <https://github.com/minio/minio-js/pull/1030> that tries to migrate querystring to use URLSearchParams.

I found that URLSearchParams is trying to replace whitespaces with '+'. There're probably more corner cases.
To keep it simple, this package <https://www.npmjs.com/package/query-string> seems to solve the problem of encoding special characters in url query

Below you can see `filename*` is now being escaped to `filename%2A` and the same for `'` character in `UTF-8%27%27`

If you try to run the tests without migrating the package, you will see <https://github.com/minio/minio-js/blob/master/src/test/functional/functional-tests.js#L924> will fail when it has special characters

---
`https://play.min.io:9000/minio-js-test-3a28c9f0-09da-4a30-8f7a-8c4ac6f1ed80/datafile-1-b?response-cache-control=No-cache&response-content-disposition=attachment%3B%20filename%3D%22t%21est.txt%22%3B%20filename%2A%3DUTF-8%27%27t%21est.txt&response-content-encoding=gzip&response-content-language=en&response-content-type=text%2Fhtml&response-expires=Sun%2C%2007%20Jun%202020%2016%3A07%3A58%20GMT&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=Q3AM3UQ867SPQQA43P2F%2F20220622%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20220622T094203Z&X-Amz-Expires=1000&X-Amz-SignedHeaders=host&X-Amz-Signature=f8b91b48f69c9f976c7451433711a4086705b5b49615a74c4f53041c16b412c3`

### CODE REVIEW

**Comment by joaofranciscosantos** [comment]
Updated package-lock

File: src/test/functional/functional-tests.js [hunk 1]
```diff
@@ -957,7 +957,7 @@ describe('functional tests', function () {
         'response-content-language': 'en',
         'response-expires': 'Sun, 07 Jun 2020 16:07:58 GMT',
         'response-cache-control': 'No-cache',
-        'response-content-disposition': 'attachment; filename=testing.txt',
+        'response-content-disposition': 'attachment; filename="t!st ng.txt"; filename*=UTF-8\'\'t&21st&20ng.txt',
```
Comment (prakashsvmx) [inline src/test/functional/functional-tests.js:960]: Please create a new test case for special characters.

**Review by prakashsvmx — CHANGES_REQUESTED** [review]

File: src/test/functional/functional-tests.js (hunk 1 above)
Comment (joaofranciscosantos) [reply src/test/functional/functional-tests.js]: done

**Review by prakashsvmx — APPROVED** [review]
Thank you. :+1:

**Review by kannappanr — APPROVED** [review]

**Pull request merged**